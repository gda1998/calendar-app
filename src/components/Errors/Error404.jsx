import { Link } from 'react-router-dom'
import { PrivateRoutes } from '../../models'
import { HeadProvider, Title } from 'react-head'
import '../../scss/components/Errors/body.scss'

const Error404 = () => {
    return (
        <>
            <HeadProvider>
                <Title>404 Not Found</Title>
            </HeadProvider>
            <div className="error">
                <div className="error__container">
                    <div className="error__head">
                        <h1>Oops!</h1>
                    </div>
                    <h2>404 - Page not found</h2>
                    <p>The page you are looking for might have been removed had its name changed or is temporarily unavailable.</p>
                    <Link to={ `/${PrivateRoutes.AUTH}/${PrivateRoutes.CALENDAR}` }>Go To Homepage</Link>
                </div>
            </div>
        </>
    );
}
export default Error404