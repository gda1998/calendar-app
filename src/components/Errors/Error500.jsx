import PropTypes from 'prop-types'
import { HeadProvider, Title } from 'react-head'
import '../../scss/components/Errors/body.scss'

const Error500 = ({ message }) => {
    return (
        <>
            <HeadProvider>
                <Title>500 Server Error</Title>
            </HeadProvider>
            <div className="error">
                <div className="error__container">
                    <div className="error__head">
                        <h1>500</h1>
                    </div>
                    <h2>500 - Server Error</h2>
                    <p>Something went wrong.</p>
                    {
                        message !== '' && <span>Error: { message }</span>
                    }
                </div>
            </div>
        </>
    );
}
Error500.propTypes = {
    message: PropTypes.string.isRequired
}
Error500.defaultProps = {
    message: ''
}
export default Error500