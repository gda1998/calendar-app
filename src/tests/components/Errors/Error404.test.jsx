import { render, screen } from '@testing-library/react';
import { Error404 } from '../../../components/Errors';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

describe('Tests in <Error404 />', () => { 
    beforeEach(() => {
        render(
            <MemoryRouter initialEntries={['/']}>
                <Routes>
                    <Route path="/" element={ <Error404 /> } />
                </Routes>
            </MemoryRouter>
        );
    });

    test('should render the component', () => { 
        expect(screen.container).toMatchSnapshot();
    });

    test('should H2 tag have the correct text', () => { 
        const h2 = screen.getByRole('heading', { level:2 });
        expect(h2.innerHTML).toContain('404 - Page not found');
    });
});