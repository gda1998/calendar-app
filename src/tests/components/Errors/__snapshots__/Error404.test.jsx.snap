// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Error404 /> should render the component 1`] = `undefined`;
